scripts/generate.py

```diff
@@ -0,0 +1,381 @@
+    with open(KRAFTCONFIG, "r", encoding="utf8") as stream:
+        data = yaml.safe_load(stream)
+        for target in data['targets']:
+            generate_build_kraft(target['name'], target['platform'])
+
+def generate_run_fc(run, arch, fs):
+    """Generate running scripts using Firecracker.
+
+    Scripts are generated in scripts/run/ directory and start
+    with the `fc-` prefix. The corresponding JSON file is generated.
+    """
+
+    json_name = os.path.join(RUN, f"fc-{arch}-{fs}-{run['name']}.json")
+    with open(json_name, "w", encoding="utf8") as stream:
+        stream.write("{\n")
+        stream.write("  \"boot-source\": {\n")
+        stream.write(f"    \"kernel_image_path\": \"workdir/build/elfloader_fc-{arch}\",\n")
+        if run["networking"]:
+            stream.write(f"    \"boot_args\": \"elfloader_fc-{arch} ")
+            stream.write("netdev.ipv4_addr=172.44.0.2 netdev.ipv4_gw_addr=172.44.0.1 ")
+            stream.write(f"netdev.ipv4_subnet_mask=255.255.255.0 -- {run['command']}\"")
+        else:
+            stream.write(f"    \"boot_args\": \"elfloader_fc-{arch} {run['command']}\"")
+        if "rootfs" in run.keys():
+            stream.write(",\n    \"initrd_path\": \"rootfs.cpio\"")
+        stream.write("\n  },\n")
+        stream.write(f"""  "drives": [],
+  "machine-config": {{
+    "vcpu_count": 1,
+    "mem_size_mib": {run['memory']},
+    "smt": false,
+    "track_dirty_pages": false
+  }},
+  "cpu-config": null,
+  "balloon": null,
+""")
+        if run["networking"]:
+            stream.write("""  "network-interfaces": [
+    {
+      "iface_id": "net1",
+      "guest_mac":  "06:00:ac:10:00:02",
+      "host_dev_name": "tap0"
+    }
+  ],
+""")
+        stream.write("""  "vsock": null,
+  "logger": {
+    "log_path": "/tmp/firecracker.log",
+    "level": "Debug",
+    "show_level": true,
+    "show_log_origin": true
+  },
+  "metrics": null,
+  "mmds-config": null,
+  "entropy": null
+}
+""")
+
+    header = TEMPLATE_RUN_FIRECRACKER_HEADER.format(json_name)
+    out_file = os.path.join(RUN, "fc-{arch}-{fs}-{run['name']}.sh")
+    with open(out_file, "w", encoding="utf8") as stream:
+        stream.write(header)
+        if "rootfs" in run.keys():
+            stream.write("\nrootfs=\"{run['rootfs']}\"\n")
+        stream.write(RUN_KILL_COMMANDS)
+        if "rootfs" in run.keys() and fs == "initrd":
+            stream.write(RUN_CPIO_COMMANDS)
+        if run["networking"]:
+            stream.write(RUN_COMMON_NET_COMMANDS)
+            stream.write(RUN_FIRECRACKER_NET_COMMANDS)
+        stream.write(RUN_FIRECRACKER_TRAILER)
+        st = os.stat(out_file)
+        os.chmod(out_file, st.st_mode | stat.S_IEXEC)
+
+def generate_run_qemu(run, arch, fs):
+    """Generate running scripts using Firecracker.
+
+    Scripts are generated in scripts/run/ directory and start
+    with the `qemu-` prefix.
+    """
+
+    kernel = os.path.join(os.path.join("workdir", "build"), f"elfloader_qemu-{arch}")
```

Comment:
The kernel for aarch64 is named `{APPNAME}_qemu-arm64`, not `{APPNAME}_qemu-aarch64`.

Reply:
The defconfigs will be renamed to use `arm64` for consistency. If we decide to make a change, we will apply it everywhere. For now, we stick with `arm64` in all places.